### PR TITLE
[W5][W10-4]Kerryn Eer

### PR DESCRIPTION
**Pull request by KerrynEer**
Add a new sort command to sort names by alphabetical order

Ready for review

### CODE REVIEW

**Comment by mmdlow** [comment]
Hi Kerryn, everything looks good for this PR. Very thorough updates for both I/O and JUnit testing

**Comment by yichong96** [comment]
Hi kerryn, very nice step by step commits that document your journey in implementing this new command. Keep it up !

File: src/seedu/addressbook/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            + ": Sorts all persons in the address book by alphabetical order temporarily.\n"
+            + "Example: " + COMMAND_WORD;
+
+
+    @Override
+    public CommandResult execute() {
+        
+        List<Person> allPersons = addressBook.getAllPersons().mutableListView();
+
+        Collections.sort(allPersons, new Comparator() {
```
Comment (kylase-learning) [inline src/seedu/addressbook/commands/SortCommand.java:27]: Is there a better way to implement the sorting in an OOP fashion? This is more like FP.

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 2]
```diff
@@ -12,17 +12,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (kylase-learning) [inline test/java/seedu/addressbook/parser/ParserTest.java:15]: According to the [Google Style Guide which we are using for this course](https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports), wildcard imports are not used.

**Review by kylase-learning — COMMENTED** [review]
Great to see you have submitted your changes via your own branch and done the documentation. 👍

You have demonstrated the progress of this exercise through the commits you have done. Maybe, as you get more comfortable, your commits can be done in batch, by committing the test, the implementation and the relevant documentation together.  instead of committing changes in each file separately. This way, it is easier to navigate through commits in events you need to revert to certain state of the code, and all its related dependencies (such as documentation, tests) are in sync. Otherwise, for example, you can revert the commit for the code change but not the documentation (if the documentation is written prior to the code).

You might want to try to write the test first before implementing the functionality.

What if the user wants to sort in the reverse order?

Please close this PR after reading comments.

File: test/java/seedu/addressbook/parser/ParserTest.java (hunk 2 above)
Comment (KerrynEer) [reply test/java/seedu/addressbook/parser/ParserTest.java:15]: Ah yes I overlooked this, will be more careful next time! 😅

Comment (KerrynEer) [reply test/java/seedu/addressbook/parser/ParserTest.java:15]: Ah yes I overlooked this, will be more careful next time! 😅

File: src/seedu/addressbook/commands/SortCommand.java (hunk 1 above)
Comment (KerrynEer) [reply src/seedu/addressbook/commands/SortCommand.java:27]: Yep I could have let Person extend the Comparable class instead! Thanks for pointing this out :)

**Pull request closed without merging**